### PR TITLE
Release and publish helm charts via a Github Action Release workflow.

### PR DESCRIPTION
**Pull request by robholland**
## What was changed

Added a new workflow that automates the release of the Helm chart. This workflow also makes the Helm chart releases available via a helm repo: https://go.temporal.io/helm-charts.

Installation via the helm repo is not yet documented, a follow-up PR will add the repo to the documentation.

## Why?

This automates some of the manual work currently performed by OSS team. This also allow users to install Temporal via Helm without having to download the repo, although they will probably still need to refer to the (rather long) values files to achieve their desired configuration.

Fixes: https://github.com/temporalio/helm-charts/issues/330

Also technically fixes: https://github.com/temporalio/helm-charts/issues/45 https://github.com/temporalio/helm-charts/issues/229, but documentation needs to be updated to show users how to install from a repo.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?
Docs will need updating to highlight the new installation method.

### CODE REVIEW

File: .github/workflows/publish-charts.yml [hunk 1]
```diff
@@ -6,13 +6,71 @@ on:
...
+          else
+            let PATCH+=1
+          fi
+          NEW_VERSION="$MAJOR.$MINOR.$PATCH"
+
+          sed -i -e "s/^version: .*/version: $NEW_VERSION/" Chart.yaml
+          sed -i -e "s/^appVersion: .*/appVersion: ${{ github.event.inputs.version }}/" Chart.yaml
+
+          git commit -m "Update Chart to $NEW_VERSION, Temporal v${{ github.event.inputs.version }}" Chart.yaml
+          git push origin
```
Comment (dandavison) [inline .github/workflows/publish-charts.yml:68]: This is really nice; I learned a couple of things from it; `grep -m` and I didn't know about the `let` arithmetic at all, I thought one had to use `((...))`. Could use `sed -E` to make the regex more readable?)

**Review by dandavison — APPROVED** [review]
LGTM!

**Comment by fakeNews-jpg** [comment]
@robholland , for information, when i try https://go.temporal.io/helm-charts i have the following error when trying to install temporal.io : "gzip: invalid header"
To reproduce: get a rancher, add this repo, go to app catalog, install the app

Do you have an idea why it does not work? 

Thanks a lot again for your amazing work :)

**Comment by fakeNews-jpg** [comment]
I just tried again when i saw the new version 0.33, and this time it seems to work at first sight ! thanks a lot @robholland  :)

**Comment by fakeNews-jpg** [comment]
I just have the issue https://github.com/temporalio/helm-charts/issues/445 popping when trying to use an existing postgresql on my helm

**Comment by wrbbz** [comment]
Are there any plans to add a folow-up documentation with Helm repo usage?

**Comment by robholland** [comment]
Yep, it's on my list: https://github.com/temporalio/helm-charts/issues/458

**Pull request merged**